File: vortexdbg-api/src/main/java/com/vortexdbg/memory/MemoryBlock.java
package com.vortexdbg.memory;

import com.vortexdbg.pointer.VortexdbgPointer;
import com.sun.jna.Pointer;

public interface MemoryBlock {

    VortexdbgPointer getPointer();

    boolean isSame(Pointer pointer);

    void free();

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/memory/MemoryBlock.kt
package com.vortexdbg.memory

import com.vortexdbg.pointer.VortexdbgPointer
import com.sun.jna.Pointer

interface MemoryBlock {

    fun getPointer(): VortexdbgPointer

    fun isSame(pointer: Pointer): Boolean

    fun free()

}
